.github/workflows/contracts.yml

```diff
@@ -0,0 +1,32 @@
+---
+name: contracts
+on:
+  push:
+jobs:
+  tests:
+    defaults:
+      run:
+        working-directory: ./packages/contracts
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [16.x]
```

Comment:
no need to update here, I'll prob do this myself after I merge, but may be worth adding 14.x since that's the latest Node version Vercel supports

Reply:
nice point. this is only testing the contracts (no Vercel) but still good practice. wdyt of setting `engines` inside all the package.json's to lock whole app to 14?

Reply:
I also just discovered `only-allow-pnpm`! I will coy over from my other repo and submit a PR

Reply:
> nice point. this is only testing the contracts (no Vercel) but still good practice. wdyt of setting `engines` inside all the package.json's to lock whole app to 14?

The way this is set up does build contracts from Vercel, so that we don't have to check in artifacts and build files:
https://github.com/holic/web3-dapp-scaffold/blob/main/packages/contracts/package.json#L5
(I am still torn on whether to check in artifacts/build files vs. building them during deploy)